core/table.ts

```diff
@@ -1,63 +1,165 @@
-import {
-  IColumnsDescriptor,
-  mapToColumnProtoArray,
-  Resolvable,
-  Session
-} from "@dataform/core/session";
+import { mapToColumnProtoArray, Session } from "@dataform/core/session";
 import * as utils from "@dataform/core/utils";
 import { dataform } from "@dataform/protos";
-
-export enum TableTypes {
-  TABLE = "table",
-  VIEW = "view",
-  INCREMENTAL = "incremental",
-  INLINE = "inline"
+import {
+  IColumnsDescriptor,
+  ICommonContext,
+  ICommonOutputConfig,
+  Resolvable
+} from "@dataform/core/common";
+import { Contextable } from "@dataform/core/contextable";
+
+/**
+ * @hidden
```

Comment:
it doesn't seem like this is being obeyed? i don't mind it being documented, but let's keep "inline" out please